pallets/anchors/src/lib.rs

```diff
@@ -148,30 +191,20 @@ decl_module! {
         /// account would be charged accordingly for the storage period.
         /// For a more detailed explanation refer section 3.4 of
         /// [Centrifuge Protocol Paper](https://staticw.centrifuge.io/assets/centrifuge_os_protocol_paper.pdf)
-        ///
-        /// # <weight>
-        /// State rent takes into account the storage cost depending on `stored_until_date`.
-        /// Otherwise independant of the inputs. The weight cost is important as it helps avoid DOS
-        /// using smaller `stored_until_date`s. Computation cost involves timestamp calculations
-        /// and state rent calculations, which we take here to be equivalent to a transfer transaction.
-        /// # </weight>
-        #[weight = 190_000_000]
-        pub fn commit(origin, anchor_id_preimage: T::Hash, doc_root: T::Hash, proof: T::Hash, stored_until_date: T::Moment) -> DispatchResult {
+        #[pallet::weight(<T as pallet::Config>::WeightInfo::commit())]
+        pub fn commit(origin: OriginFor<T>, anchor_id_preimage: T::Hash, doc_root: T::Hash, proof: T::Hash, stored_until_date: T::Moment) -> DispatchResult {
             let who = ensure_signed(origin)?;
             // validate the eviction date
             let eviction_date_u64 = TryInto::<u64>::try_into(stored_until_date)
                 .map_err(|_e| "Can not convert eviction date to u64")
                 .unwrap();
```

Comment:
Doesn't that what `.map_err` does ?

Reply:
`.map_err` just changes the error from whatever `::try_into` returns into the string error. `unwrap` will still panic. Instead of `unwrap`, we should use the `?` operator here to do an early return.

Reply:
Since this is code from before the refactor, I'm inclined to just have somebody throw up a separate "Fix potential panics" PR to not block this cleanup?